src/libraries/System.Drawing.Common/tests/ImageTests.cs

```diff
@@ -157,6 +157,23 @@ public static IEnumerable<object[]> GetEncoderParameterList_ReturnsExpected_Test
                     new Guid(unchecked((int)0xa219bbc9), unchecked((short)0x0a9d), unchecked((short)0x4005), new byte[] { 0xa3, 0xee, 0x3a, 0x42, 0x1b, 0x8b, 0xb0, 0x6c }) /* Encoder.SaveAsCmyk.Guid */
                 }
             };
+
+#if !NETFRAMEWORK
```

Comment:
Nit: for tests and test data, we usually use, if (!PlatformDetection.IsFullFramework)

Reply:
Hmm,

```csharp
ImageTests.cs(161,36): error CS0117: (NETCORE_ENGINEERING_TELEMETRY=Build) 'PlatformDetection' does not contain a definition for 'IsFullFramework'
```

What am I missing here?

Reply:
`IsFullFramework`, obivously :)

It was removed in https://github.com/dotnet/corefx/pull/37972.

Reply:
Ugh 😄 .

I'm a bit confused, because dotnet/corefx#37972 and dotnet/runtime#28619 seem to indicate tests would no longer run on full framework, yet they do for System.Drawing.Common.

@safern What's the way forward here?

Reply:
The test should run on `netfx` only if the library targets `netstandard2.0`. In that case they should test the functionality of the `netstandard2.0` library running on NetFX, not testing NetFX code itself.

Reply:
The tricky issue is the fact that the NuGet package contains `lib/net461` implementation that forwards all the types to `System.Drawing`. I am not entirely sure how is that produced but it is likely problematic going forward.

Reply:
> IsFullFramework, obviously :)
> It was removed in dotnet/corefx#37972.

It was not removed, it was renamed to `IsNetFramework`. https://github.com/dotnet/runtime/blob/0b25063ec056b0d6668bf0848d108d8c873f3fce/src/libraries/Common/tests/CoreFx.Private.TestUtilities/System/AssertExtensions.cs#L16

https://github.com/dotnet/runtime/pull/1900

> The test should run on netfx only if the library targets netstandard2.0. In that case they should test the functionality of the netstandard2.0 library running on NetFX, not testing NetFX code itself.

Not really. We want to run tests on `netfx` for libraries that ship a netfx asset. In this case even if it is a facade, we want to test that we're producing the right facade and that tests run correctly.

Reply:
> Not really. We want to run tests on netfx for libraries that ship a netfx asset. 

Fair enough. I thought these ones are mostly in frozen state now and slowly removed from repository with no forward maintenance plan except from release branches.

> In this case even if it is a facade, we want to test that we're producing the right facade and that tests run correctly.

It's somewhat weird to have new APIs that are not present on certain platform. What will happen if you call them? I'd except something like `PlatformNotSupportedException` but that only works for new types and only if partial facade is generated. In this case it's new members on existing types.

Reply:
> It's somewhat weird to have new APIs that are not present on certain platform. What will happen if you call them? I'd except something like PlatformNotSupportedException but that only works for new types and only if partial facade is generated. In this case it's new members on existing type.

You will not be able to reference them anymore as for ref we only build live for netcoreapp3.0 and netcoreapp5.0. The netfx assets harvested. So now that you mention it we no longer build netfx assets for System.Drawing.Common anymore as we just get them from the latest package and re-ship  them in the new package. Maybe it is fine removing the netfx tests for this library.

cc: @ericstj @ViktorHofer @danmosemsft  @JeremyKuhne -- I'm leaning towards removing testing netfx for S.Drawing.Common. What do you guys think?

Reply:
> What do you guys think?

My only concern is that surfacing unintentional deviation in behavior from Framework becomes more difficult without runs there. If we had better test coverage I'd worry less, but I think we're pretty far from adequately covering GDI+ behavior in our current tests.

Reply:
Yeah that's a fair point. I would suggest leaving the tests on this PR and using the `NetFramework` property in `PlatformDetection`; then we can discuss further if we want to stop running tests for netfx in this project.

Reply:
So, unfortunately we went the full circle on this one - just using `PlatformDetection` does not work, because there's a reference to `Encoder.ImageItems` which was added as part of this PR, but is not present in NetFx.

Reply:
You're right. Ok, well this at least was a good discussion. Then we will need to go back to the compile time condition, sorry about that.